### PR TITLE
feat: add welcome flow publish page.

### PR DESCRIPTION
**Pull request by yashdesu**
#### Description of the change
This change adds in the publish page (part of the welcome flow) with the required smart-contract updations and UI/UX state management.

#### Checklist

**I agree to the following :-**

- [x] Added description of the change
- [x] I've read the [contributing guidelines](https://github.com/WebXDAO/devprotocol.xyz/blob/main/CONTRIBUTING.md)
- [x] Search previous suggestions before making a new PR, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

**Notes**: This change needs to be tested. Due to lack of funds, while integrating smart-contract updation features (initialize membership and set membership intersteps) testing was not done.

### CODE REVIEW

File: src/components/PublishNetworkSelection/PublishNetworkSelection.vue [hunk 1]
```diff
@@ -182,8 +221,15 @@
...
 import type Web3Modal from 'web3modal'
 import { defineComponent } from '@vue/runtime-core'
+import { clientsSTokens } from '@devprotocol/dev-kit/agent'
 import type { connection as Connection } from '@devprotocol/clubs-core/connection'
-import { add } from 'ramda'
+import { BaseProvider } from '@ethersproject/providers'
+import {
+  address,
+  callSimpleCollections,
+} from '@plugins/memberships/utils/simpleCollections'
```
Comment (stuartwk) [inline src/components/PublishNetworkSelection/PublishNetworkSelection.vue:230]: I feel like importing functions from plugins into components seems like an anti-pattern? Maybe we could move these into fixtures? Would be fine to do this in a separate PR if it makes things easier.

Comment (aggre) [reply src/components/PublishNetworkSelection/PublishNetworkSelection.vue:230]: In the future, plugins can be separated out as npm packages, so importing external plugins in this manner would be acceptable. (but we must not forget that plugins export those functions)

File: src/components/PublishNetworkSelection/PublishNetworkSelection.vue [hunk 2]
```diff
@@ -422,9 +425,13 @@ export default defineComponent({
...
-          requiredETHAmount: opt.price, // TODO: confirm that this is in eth always.
-          requiredETHFee: opt.fee, // TODO: confirm that this is in eth always.
-          gateway: '', // TODO: Where does this come from?
+          requiredETHAmount: parseUnits(String(opt.price)).toString(),
+          requiredETHFee: opt.fee?.percentage
+            ? parseUnits(
+                new BigNumber(opt.price).times(opt.fee.percentage).toFixed()
+              ).toString()
+            : 0,
+          gateway: '', // TODO: Where does this come from? <= this is a connected wallet address
```
Comment (yashdesu) [inline src/components/PublishNetworkSelection/PublishNetworkSelection.vue:434]: I've fetched the currentWalletAddress while component is mounting and used it over here.

**Review by aggre — APPROVED** [review]
I've added some tweaking
